include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -1039,8 +1504,100 @@ namespace internal
                    is_src_locally_owned,
                    false);
     }
+
+    virtual ~PermutationFineDoFHandlerView() = default;
   };
 
+
+
+  template <int dim, int spacedim>
+  bool
+  p_transfer_without_repartitioning(
+    const DoFHandler<dim, spacedim> &dof_handler_fine,
+    const DoFHandler<dim, spacedim> &dof_handler_coarse,
+    const unsigned int               mg_level_fine,
+    const unsigned int               mg_level_coarse)
+  {
+    if (mg_level_fine != mg_level_coarse)
+      return false;
+
+    if (&dof_handler_fine.get_triangulation() !=
+        &dof_handler_coarse.get_triangulation())
+      return false;
+
+    return true;
+  }
+
+
+
+  template <int dim, int spacedim>
+  bool
+  h_transfer_with_first_child_policy(
```

Comment:
Again, I would prefer to have a verb, but maybe it is just me - feel free to ignore this one
```suggestion
  h_transfer_uses_first_child_policy(
```